saleor/order/utils.py

```diff
@@ -234,25 +227,18 @@ def update_taxes_for_order_lines(
     )
 
 
-def update_order_prices(order: Order, manager: "PluginsManager", tax_included: bool):
+def update_order_prices_if_expired(
+    order: Order, manager: "PluginsManager", tax_included: bool
+):
     """Update prices in order with given discounts and proper taxes."""
 
-    update_taxes_for_order_lines(order.lines.all(), order, manager, tax_included)
+    lines = order.lines.all()
+    update_order_lines_prices_if_expired(lines, order, manager, tax_included)
 
     if order.shipping_method:
-        shipping_price = manager.calculate_order_shipping(order)
-        order.shipping_price = shipping_price
-        order.shipping_tax_rate = manager.get_order_shipping_tax_rate(
-            order, shipping_price
-        )
-        order.save(
-            update_fields=[
-                "shipping_price_net_amount",
-                "shipping_price_gross_amount",
-                "shipping_tax_rate",
-                "currency",
-            ]
-        )
+        # recalculate order.shipping_price and order.shipping_tax_rate
+        fetch_order_prices_if_expired(order, manager)
+        order.save(update_fields=["currency"])
```

Comment:
Is this save needed? A save is already done in `fetch_order_prices_if_expired`. Also, what's the use-case with saving the currency, can it change here?